app/Commands/Session/EndCommand.php

```
@@ -0,0 +1,174 @@
+        $gitRoot = shell_exec('git rev-parse --show-toplevel 2>/dev/null');
+        if ($gitRoot !== null && trim($gitRoot) !== '') {
+            return basename(trim($gitRoot));
+        }
+
+        return basename($cwd);
+    }
+
+    private function detectBranch(): ?string
+    {
+        $branch = shell_exec('git branch --show-current 2>/dev/null');
+        if ($branch !== null && trim($branch) !== '') {
```

PHPStan (2.1.33)

Parameter #1 $string of function trim expects string, string|false given.


(argument.type)